luxjs/forms/utils.js

```diff
@@ -1,40 +1,59 @@
 /**
  * Created by Reupen on 02/06/2015.
+ * Edited by Tom on 18/12/2015.
  */
 
-angular.module('lux.form.utils', ['lux.services'])
+angular.module('lux.form.utils', ['lux.pagination'])
 
-    .directive('remoteOptions', ['$lux', function ($lux) {
+    .directive('remoteOptions', ['$lux', 'luxPaginationFactory', function ($lux, LuxPagination) {
 
-        function fill(api, target, scope, attrs) {
+        function remoteOptions(luxPag, target, scope, attrs, element) {
 
-            var id = attrs.remoteOptionsId || 'id',
-                nameOpts = attrs.remoteOptionsValue ? JSON.parse(attrs.remoteOptionsValue) : {
-                    type: 'field',
-                    source: 'id'
-                },
-                nameFromFormat = nameOpts.type === 'formatString',
-                initialValue = {},
-                params = JSON.parse(attrs.remoteOptionsParams || '{}'),
-                options = [];
+            function lazyLoad(e) {
+                // lazyLoad requests the next page of data from the API
+                // when nearing the bottom of a <select> list
+                var uiSelect = element[0].querySelector('.ui-select-choices');
 
-            scope[target.name] = options;
+                e.stopPropagation();
+                if (!uiSelect) return;
 
-            initialValue.id = '';
-            initialValue.name = 'Loading...';
+                var uiSelectChild = uiSelect.querySelector('.ui-select-choices-group');
+                uiSelect = angular.element(uiSelect);
 
-            options.push(initialValue);
+                uiSelect.bind('scroll', function() {
```

Comment:
I think `on` and `off` are preferred to `bind` and `unbind` (unless they don't work for some reason).